app/components/amount-input-screen/amount-input-screen.tsx

```diff
@@ -198,11 +200,32 @@ export const AmountInputScreen: React.FC<AmountInputScreenProps> = ({
     [currencyInfo],
   )
 
-  const onToggleCurrency =
-    secondaryNewAmount &&
-    (() => {
+  const client = useApolloClient()
+  const swapDefaultUnitOfAccount =
+    useSwapDefaultUnitOfAccountQuery().data?.swapDefaultUnitOfAccount
+  const [needSwapCurrencyInit, setNeedSwapCurrencyInit] = React.useState(
+    swapDefaultUnitOfAccount,
+  )
+
+  const onToggleCurrency = React.useCallback(() => {
+    if (secondaryNewAmount) {
+      setSwapDefaultUnitOfAccount(client, !swapDefaultUnitOfAccount)
       setNumberPadAmount(secondaryNewAmount)
-    })
+    }
+  }, [secondaryNewAmount, setNumberPadAmount, client, swapDefaultUnitOfAccount])
+
+  useEffect(() => {
```

Comment:
Does this cause a flicker allowing it to render the initial amount and then on the next render switching to the default? I wonder if the change needs to be made before getting to this component, so that the first time it renders it is rendering the correct money amount.

Reply:
I have not seen any flicker in my simulator. I think there were to do some calculation before rendering but I don't remember what it was anymore, I'll look for it

Reply:
nothing obvious coming to mind from my convo with GPT4. let's merge and see how it behaves and this is an issue?